### PR TITLE
Correctly capture the subscriber for timeout

### PR DESCRIPTION
**Pull request by jcf120**
A promise with N subscribers was only firing the last in the queue, and
was firing the same subscriber N times.

### CODE REVIEW

File: src/Promise.js [hunk 1]
```diff
@@ -154,23 +154,29 @@ if (!window.Promise) {
...
+                    sub.resolveCallback(value);
+                    promise.settledValue = value;
+                };
+            }
             var promise = this, subscriber;
             while (this.subscriberqueue.length) {
                 subscriber = this.subscriberqueue.shift();
-                setTimeout(function () {
-                    subscriber.resolveCallback(value);
-                    promise.settledValue = value;
```
Comment (jcf120) [inline src/Promise.js:162]: Does it matter that the settled value is reassigned per subscriber you reckon?

**Comment by shortercode** [comment]
ahh darn you and your lack of block scoping ES5... I'm actually started working on a DRYer / closer to spec rewrite on friday night that should have fixed this issue. Currently sitting on a train so will try to finish it up and will add a pull for your review

**Comment by jcf120** [comment]
Okey cool. Ta!

File: src/Promise.js (hunk 1 above)
Comment (shortercode) [reply src/Promise.js:162]: nah it just sets it to the same value each time, kinda pointless but not dangerous

**Comment by shortercode** [comment]
okay so most of the refactor is mostly done, and its looking pretty good but I got kinda stuck trying to remember the finer details of value settlement in recursive promises... lets merge this for now and the new one can go in when I'm sure it's right!

**Comment by jcf120** [comment]
Fair enough!

**Pull request merged**